tests/neg/i3630.scala

```diff
@@ -0,0 +1 @@
+class U { Array(Array(2), Array("a")) } // error: no classtag
```

Comment:
Do we really need to disallow this? It seems to work OK in scalac.

Reply:
The full error message says:

    No ClassTag available for Array[_ >: Int & String <: Int | String]

The inferred type of the outer array is

    Array[Array[_ >: Int & String <: Int | String]]

That is indeed the best type. But there's no ClassTag. We *could* infer some wider type such as 

    Array[AnyRef](Array(2), Array("a"))

for the outer array. But it would be a special case. And I am not sure whether in the end it would not be better to reject this weird construction as an error and demand that the user write

    Array[AnyRef](Array(2), Array("a"))

explicitly.